kDrive/UI/Controller/Files/Categories/ManageCategoriesViewController.swift

```diff
@@ -252,8 +252,13 @@ final class ManageCategoriesViewController: UITableViewController {
             return
         }
         self.driveFileManager = driveFileManager
-        let realm = driveFileManager.getRealm()
-        files = filesId.compactMap { driveFileManager.getCachedFile(id: $0, using: realm) }
+
+        let matchedFiles = driveFileManager.fetchResults(ofType: File.self) { faultedCollection in
+            faultedCollection.filter("id IN %@", filesId)
+        }
+
+        files = matchedFiles.compactMap { !$0.isInvalidated ? $0 : nil }
```

Comment:
Still not convince by this, from my point of view objects cannot be invalidated right after being pulled out of a realm. I will do some tests on my side.

Reply:
Yes. Since this comes from a throw away realm / Transactionable, it is fine without it.